lib/import/brca/providers/st_george/st_george_handler.rb

```diff
@@ -1,436 +1,348 @@
+            # interrogate variant dna column
+            if record.raw_fields['variant dna'].present?
+              interrogate_variant_dna_column(record, genotype, genes, column, gene)
+            # interrogate raw gene (other)
+            elsif /fail/i.match(record.raw_fields['gene (other)']).present?
+              return if match_fail(gene, record, genotype)
 
-          def process_uniq_gene_rec(record, positive_genes)
-            if record.raw_fields['genotype'].scan(DELIMETER_REGEX).size.positive?
-              variants = process_split_variants(record, positive_genes)
+              update_status(10, 1, column, 'gene', genotype)
+            elsif record.raw_fields['gene (other)']&.match(/\?\z/ix)
+              genotype.add_status(4)
+            elsif record.raw_fields['gene (other)']&.match(/^c\.|^Ex.*Del\z|^Ex.*Dup\z|^Het\sDel|^Het\sDup/ix)
+              update_status(2, 1, column, 'gene', genotype)
             else
-              positive_genes *= record.raw_fields['genotype'].scan(CDNA_REGEX).
-                                flatten.compact.size
-              variants = process_single_variant(record, positive_genes)
-            end
-            variants
-          end
+              genotype.add_status(4)
 
-          def process_single_variant(record, positive_genes)
-            mutation = get_cdna_mutation(record.raw_fields['genotype'])
-            protein = get_protein_impact(record.raw_fields['genotype'])
-            positive_genes.zip(mutation, protein.flatten.compact)
+              gene_classv_gene_n_format(record, genotype, gene)
+            end
           end
 
-          def process_split_variants(record, positive_genes)
-            record.raw_fields['genotype'].scan(DELIMETER_REGEX)
-            raw_genotypes = record.raw_fields['genotype'].split($LAST_MATCH_INFO[0])
-            variants = []
-            raw_genotypes.each do |raw_genotype|
-              if positive_genes == []
-                positive_gene_rec = []
-                gene = raw_genotype.scan(BRCA_GENES_REGEX)
-                deprec_gene = raw_genotype.scan(DEPRECATED_BRCA_NAMES_REGEX)
-                process_rightname_gene(gene, positive_gene_rec) if gene.present?
-                process_deprecated_gene(deprec_gene, positive_gene_rec) if deprec_gene.present?
+          def interrogate_variant_dna_column(record, genotype, genes, column, gene)
+            # For full screen tests only- add test status when variant dna column is not empty
+            if record.raw_fields['variant dna'].match(/Fail/ix)
+              genotype.add_status(9)
+            elsif record.raw_fields['variant dna'] == 'N'
+              genotype.add_status(1)
+            # variant dna is not '*Fail*', 'N' or null AND raw:gene is not null AND raw:gene (other) is null
+            # 2 (abnormal) for gene in raw:gene. 1 (normal) for all other genes.
+            elsif record.raw_fields['gene'].present? \
```

Comment:
You don't need backslashes for continuation here (and throughout the file) , it can be written as -

record.raw_fields['gene'].present? &&
record.raw_fields['gene (other)'].blank?

Reply:
If I remove the backslashes my tests fail and I get "syntax error, unexpected &&"

Reply:
@lauramccluskey1 Did you move `&&` operator to first line ?, below will work

```
def interrogate_variant_dna_column(record, genotype, genes, column, gene)
            # For full screen tests only- add test status when variant dna column is not empty
            if record.raw_fields['variant dna'].match(/Fail/ix)
              genotype.add_status(9)
            elsif record.raw_fields['variant dna'] == 'N'
              genotype.add_status(1)
            # variant dna is not '*Fail*', 'N' or null AND raw:gene is not null AND raw:gene (other) is null
            # 2 (abnormal) for gene in raw:gene. 1 (normal) for all other genes.
            elsif record.raw_fields['gene'].present? && record.raw_fields['gene (other)'].blank?
              update_status(2, 1, column, 'gene', genotype)
            # variant dna is not '*Fail*', 'N' or null AND raw:gene is not null AND raw:gene (other) is not null
            # 2 (abnormal) for gene in raw:gene.
            # 9 (failed, genetic test) for any gene specified WITH 'Fail' in raw:gene (other).
            # 1 (normal) for all other genes
            elsif record.raw_fields['gene'].present? && record.raw_fields['gene (other)'].present?
              if column == 'gene'
                genotype.add_status(2)
              elsif column == 'gene (other)'
                match_fail(gene, record, genotype)
              else
                genotype.add_status(1)
              end
            # variant dna not '*Fail*', 'N' or null AND raw:gene is null AND raw:gene(other) not a single gene
            # If gene is specified in raw:variant dna, assign 2 (abnormal) for that gene and 1 (normal) for
            # all other genes.
            # Else interrogate raw:gene (other).
            elsif record.raw_fields['gene'].blank? && (genes['gene (other)'].blank? ||
                  genes['gene (other)'].length > 1) && !genes['variant dna'].nil? && 
                  genes['variant dna'].length >= 1
              update_status(2, 1, column, 'variant dna', genotype)
            # variant dna is not '*Fail*', 'N' or null AND raw:gene is null AND raw:gene (other) specifies a single gene
            # 2 (abnormal) for gene in raw:gene (other). 1 (normal) for all other genes.
            elsif record.raw_fields['gene'].blank? && !genes['gene (other)'].nil? && genes['gene (other)'].length == 1
              update_status(2, 1, column, 'gene (other)', genotype)
            end
          end
```

Reply:
Thanks @shilpigoeldev - updated in [960d753](https://github.com/NHSDigital/data_management_system/pull/77/commits/960d753ac2d56ab6510a5e6b45508ca55d85128d)